packages/jest-cli/src/runGlobalHook.js

```diff
@@ -59,20 +64,7 @@ export default ({
       );
 
       // $FlowFixMe
-      let globalModule = require(modulePath);
-
-      if (
-        typeof globalModule === 'object' &&
-        globalModule.__esModule === true
-      ) {
-        if (globalModule.default) {
-          globalModule = globalModule.default;
-        } else {
-          throw new TypeError(
-            `${moduleName} file must use a default export with ES Modules at ${modulePath}`,
-          );
-        }
-      }
+      const globalModule = _interopRequireDefault(require(modulePath)).default;
```

Comment:
Honestly I think the other solution is better. It provides the real error message without having the developer debug what is going on (as obvious as we think it should be).

Reply:
If I remember correctly, jest 25 is set to use typescript config files. So another idea is to extract the original function out so it can later be used in jest 25 when written as a `.ts` file.

Reply:
we'll still be using babel at that point (and `tsc`'s module output is compatible with babel's way of detection anyways, for compat)

Reply:
FYI, this is where I originally found the ``globalModule.__esModule`` check: https://github.com/babel/babel/blob/6.x/packages/babel-core/src/transformation/file/options/option-manager.js#L277

I've since used it in my other projects, where I just copied it from.